core/domain/blog_services.py

```diff
@@ -56,7 +74,33 @@ def get_blog_post_from_model(blog_post_model):
         blog_post_model.published_on)
 
 
-def get_blog_post_by_id(blog_post_id, strict=True):
+@overload
+def get_blog_post_by_id(
+    blog_post_id: str
+) -> blog_domain.BlogPost: ...
+
+
+@overload
+def get_blog_post_by_id(
+    blog_post_id: str, strict: Literal[True]
+) -> blog_domain.BlogPost: ...
+
+
+@overload
+def get_blog_post_by_id(
+    blog_post_id: str, strict: Literal[False]
+) -> Optional[blog_domain.BlogPost]: ...
```

Comment:
I'm thinking that we should enforce that the strict arg always needs to be named argument.